src/lib/Libutil/misc_utils.c

```diff
@@ -403,6 +404,52 @@ pbs_fgets_extend(char **pbuf, int *pbuf_size, FILE *fp)
 	return buf;
 }
 
+/**
+ * @brief
+ *	Internal asprintf() implementation for use on all platforms
+ *
+ * @param[in, out] dest - character pointer that will point to allocated
+ *			  space ** must be freed by caller **
+ * @param[in] fmt - format for printed string
+ * @param[in] ... - arguments to format string
+ *
+ * @return int
+ * @retval -1 - Error
+ * @retval >=0 - Length of new string, not including terminator
+ */
+int
+pbs_asprintf(char **dest, const char *fmt, ...)
+{
+	va_list args, dupargs;
+	int len, rc = -1;
+	char c, *buf = NULL;
+
+	if (!dest)
+		return -1;
+	*dest = NULL;
```

Comment:
So this can potentially have pbs_asprintf() return a NULL to *dest. In looking at all the calls to pbs_asprintf(), there's no check for return value (e.g. pbs_asprintf(&buf, ...), so 'buf' can be NULL), and there are succeeding functions that access directly that 'buf' like schdlog(..., buf) which will crash the program. Perhaps, force pbs_asprintf() to always return an empty string (buf[0] == '\0') even when it fails? Or, will have to check return value of pbs_asprintf() across all references in the code...

Reply:
I patched this as best I could. Recall that realloc(NULL, 1) is equivalent to malloc(1).